build/set-env.ps1

```diff
@@ -6,38 +6,67 @@ $ErrorActionPreference = 'Stop'
 Write-Host "Setting up build environment variables"
 
 If ($Env:BUILD_BUILDNUMBER -eq $null) { $Env:BUILD_BUILDNUMBER ="0.0.1.0" }
+Write-Host "BUILD_BUILDNUMBER: $Env:BUILD_BUILDNUMBER"
+
 If ($Env:BUILD_CONFIGURATION -eq $null) { $Env:BUILD_CONFIGURATION ="Debug" }
+Write-Host "BUILD_CONFIGURATION: $Env:BUILD_CONFIGURATION"
+
 If ($Env:BUILD_VERBOSITY -eq $null) { $Env:BUILD_VERBOSITY ="m" }
+Write-Host "BUILD_VERBOSITY: $Env:BUILD_VERBOSITY"
+
 If ($Env:ASSEMBLY_VERSION -eq $null) { $Env:ASSEMBLY_VERSION ="$Env:BUILD_BUILDNUMBER" }
+Write-Host "ASSEMBLY_VERSION: $Env:ASSEMBLY_VERSION"
+
 If ($Env:PYTHON_VERSION -eq $null) { $Env:PYTHON_VERSION = "${Env:ASSEMBLY_VERSION}a1" }
+Write-Host "PYTHON_VERSION: $Env:PYTHON_VERSION"
+
 If ($Env:NUGET_VERSION -eq $null) { $Env:NUGET_VERSION ="0.0.1-alpha" }
+Write-Host "NUGET_VERSION: $Env:NUGET_VERSION"
 
-If (($Env:ENABLE_NATIVE -ne "false") -and ($Env:NATIVE_SIMULATOR -eq $null) ) {
-    $Env:NATIVE_SIMULATOR = (Join-Path $PSScriptRoot "..\src\Simulation\Native\build\drop")
+
+If ($Env:ENABLE_NATIVE -ne "false") {
+    If ($Env:NATIVE_SIMULATOR -eq $null) {
```

Comment:
Testing with string interpolation also allows for catching when the environment variable is defined, but is empty:
```suggestion
    If ("$Env:NATIVE_SIMULATOR" -eq "") {
```

Reply:
Hm.. Interesting.
As I understand, the case `If ("$Env:NATIVE_SIMULATOR" -eq "") {` does not catch the undefined variables. If we want to catch the undefined vars and empty vars then we need to write 
`($Env:NATIVE_SIMULATOR -eq $null) -or ("$Env:NATIVE_SIMULATOR" -eq "")`
Is that a correct understanding?

Reply:
I believe that in the context of string interpolation, undefined environment variables expand to empty strings, such that `"$Env:FOOBAR" -eq ""` evaluates to `$true` if the environment variable `FOOBAR` is either undefined or is defined as a blank string.